structural/script.js

```diff
@@ -300,6 +300,295 @@ const formula_circle_I_y_bar = [
     },
 ];
 
+const formula_half_circle_x_bar = [
+    {
+        displayName: 'Calculate x_bar',
+        calculate: (half_cir_D) => half_cir_D/2
+    },
+    {
+        displayName: 'Calculate D',
+        calculate: (half_cir_x_bar) => half_cir_x_bar*2
+    },
+];
+
+const formula_half_circle_y_bar = [
+    {
+        displayName: 'Calculate y_bar',
+        calculate: (half_cir_r) => (4*half_cir_r)/(3*Math.PI)
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_y_bar) => (3*Math.PI*half_cir_y_bar) / 4
+    },
+];
+
+const formula_half_circle_area = [
+    {
+        displayName: 'Calculate area',
+        calculate: (half_cir_r1) => (Math.PI * half_cir_r1**2) / 2
+    },
+    {
+        displayName: 'Calculate r',
+        calculate: (half_cir_area) => (2 * half_cir_area / Math.PI)**(1/2)
+    },
+];
+
+const formula_half_circle_I_x_bar = [
```

Comment:
There is something wrong with this one 
if x bar = 1 then radius = 1.7373686553786964 
if radius =  1.7373686553786964 then x bar = 9.869604401089356